docs/design/eth-subscribe.md

```diff
@@ -0,0 +1,366 @@
+# Subscription design
+
+## Purpose
+
+The JSON-RPC Relay currently doesn't support subscription using websocket as is common in many other relays and is expected by many web3 tools, which makes polling for event logs, new blocks and transactions very hard for developers.
+
+## Goals
+
+1. Implement websocket subscription in the relay
+2. Introduce two new methods - `eth_subscribe` and `eth_unsubscribe`
+3. Support subscription for event logs
+
+## Non-Goals
+
+1. Add long-term memory to the relay
+2. Reduce the efficiency of the relay or mirror node due to high resource usage
+3. Support subscription for new blocks and pending transactions
+
+## Architecture
+
+The websocket connections should be handled by its own node process, this can be done by creating a new package - `ws-server`;
+
+### Subscription Types
+
+|   Subscription Type    |                                  Description                                   |                           Support                           |
+| :--------------------: | :----------------------------------------------------------------------------: |:-----------------------------------------------------------:|
+|          logs          |      Emits logs attached to a new block that match certain topic filters.      |                       Yes, initially                        |
+|        newHeads        |               Emits new blocks that are added to the blockchain.               |                 No, maybe in future release                 |
+| newPendingTransactions | Emits transaction hashes that are sent to the network and marked as `pending`. | No, as Hedera does not have pending transactions on a node. |
+
+### Initiating a subscription
+
+#### Request
+
+```javascript
+// initiate websocket stream first
+wscat -c wss://<env>.hashio.io/api
+
+// then call subscription
+{"jsonrpc":"2.0","id": 1, "method": "eth_subscribe", "params": [SUBSCRIPTION_TYPE, PARAMS]}
+```
+
+#### Response
+
+```javascript
+{"jsonrpc":"2.0","id": 1, "result": SUBSCRIPTION_ID}
+```
+
+### Logs
+
+     The `logs` subscription type emits logs that match a specified topic filter and are included in newly added blocks.
+
+#### Parameters
+
+An object with the following fields:
+
+- address (optional): [`string`] or [`array of strings`] Singular address or array of addresses. Only logs created from one of these addresses will be emitted.
```

Comment:
As part of release we can prioritize `string` and have `array of strings` follow in a future release.
These should be separate tickets